From 0f24ebe8706a0ff9351bd4a8c50779de94750d15 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Tue, 14 Aug 2012 18:20:32 +0900
Subject: [PATCH 48/97] dfb-video-example: properly update playback rate to
 1.0 and less

When trying to change the playback rate to 1.0 or less, the playback rate
was not actually updated.
---
 ext/directfb/dfb-video-example.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/ext/directfb/dfb-video-example.c b/ext/directfb/dfb-video-example.c
index 18ab707..66fd03a 100644
--- a/ext/directfb/dfb-video-example.c
+++ b/ext/directfb/dfb-video-example.c
@@ -215,6 +215,7 @@ channel_cb (GIOChannel * source, GIOCondition condition, gpointer data)
       }
     } else if (rate > 0.0 && rate <= 1.0) {
       printf ("change playback rate to %0.5lf\n", rate);
+      playback_rate = rate;
       if (!gst_element_seek (pipeline, rate, GST_FORMAT_TIME,
               GST_SEEK_FLAG_ACCURATE | GST_SEEK_FLAG_FLUSH,
               GST_SEEK_TYPE_SET, position,
-- 
1.7.10.4

